### PR TITLE
Integrate grpc client to bridge outgoing data

### PR DESCRIPTION
**Pull request by mariusmihaic**
## Proposed changes
- Integrated grpc client to send bridge txs
- On sovereign end subround, the leader will send txs on a go routine in order not to add extra time to the overall block time

## Testing procedure
- Node logs for bridge op hash **b53aada97b616050bd4a5f39d0ea8348a814595ce4a564d56290a0d66ee56f27** :

image

```
DEBUG[2023-12-06 14:50:27.586] [consensus/spos/bls] [0/0/39/(END_ROUND)] sent outgoing operations                 hashes = [5852ce70d2a0a7eaaf4a10e9f72210c70573d95e314386f9b9d765ef7d0fc885 535222bb31d9801102f386c74f6f120e67c1487af20fd1f738a45ae772a7d498 1beca4dbcf5a1a5aa7ce90e33a0688573b7d27e320085d218ddd07faeffbbb5b] 
```

- GRPC server to send txs logs:

```
DEBUG[2023-12-06 14:50:26.542]   creating tx data                         bridge op hash = b53aada97b616050bd4a5f39d0ea8348a814595ce4a564d56290a0d66ee56f27 
DEBUG[2023-12-06 14:50:26.543]   sending bunch                            index = 0 
INFO [2023-12-06 14:50:27.585]   sent tx                                  hash = 5852ce70d2a0a7eaaf4a10e9f72210c70573d95e314386f9b9d765ef7d0fc885 
INFO [2023-12-06 14:50:27.585]   sent tx                                  hash = 535222bb31d9801102f386c74f6f120e67c1487af20fd1f738a45ae772a7d498 
INFO [2023-12-06 14:50:27.585]   sent tx                                  hash = 1beca4dbcf5a1a5aa7ce90e33a0688573b7d27e320085d218ddd07faeffbbb5b 

```

- Testnet bridge transactions:
- https://testnet-explorer.multiversx.com/transactions/5852ce70d2a0a7eaaf4a10e9f72210c70573d95e314386f9b9d765ef7d0fc885
- https://testnet-explorer.multiversx.com/transactions/535222bb31d9801102f386c74f6f120e67c1487af20fd1f738a45ae772a7d498
- https://testnet-explorer.multiversx.com/transactions/1beca4dbcf5a1a5aa7ce90e33a0688573b7d27e320085d218ddd07faeffbbb5b

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: cmd/sovereignnode/config/sovereignConfig.toml [hunk 1]
```diff
@@ -37,3 +37,7 @@
     SubscribedEvents = [
     { Identifier = "bridgeOps", Addresses = ["erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th"] }
 ]
+
+[OutGoingBridge]
+    GRPCHost = "localhost"
+    GRPCPort = "8085"
```
Comment (iulianpascalau) [inline cmd/sovereignnode/config/sovereignConfig.toml:43]: As stated, let's not open another port. Please find a solution to this.

File: consensus/spos/bls/sovereignSubRoundEnd.go [hunk 2]
```diff
@@ -110,9 +105,21 @@ func (sr *sovereignSubRoundEnd) getAllOutGoingOperations(currentOperations *sove
...
+	log.Debug("current outgoing operations", "hash", currentOperations.Hash)
 	return append(outGoingOperations, currentOperations)
 }
 
+func (sr *sovereignSubRoundEnd) sendOutGoingOperations(ctx context.Context, data []*sovereign.BridgeOutGoingData) {
+	resp, err := sr.bridgeOpHandler.Send(ctx, &sovereign.BridgeOperations{
+		Data: data,
+	})
+	if err != nil {
+		log.Error("sovereignSubRoundEnd.doSovereignEndRoundJob.bridgeOpHandler.Send", "error", err)
```
Comment (iulianpascalau) [inline consensus/spos/bls/sovereignSubRoundEnd.go:117]: missing return

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: cmd/sovereignnode/config/sovereignConfig.toml (hunk 1 above)
Comment (mariusmihaic) [reply cmd/sovereignnode/config/sovereignConfig.toml:43]: This is the port of the server. Each node will be client. No port is opened on node side.

File: consensus/spos/bls/sovereignSubRoundEnd.go (hunk 2 above)
Comment (mariusmihaic) [reply consensus/spos/bls/sovereignSubRoundEnd.go:117]: Yes, good catch.
This was however fixed on this branch already: https://github.com/multiversx/mx-chain-go/pull/5763/files#diff-e6295dd6812e5903f1738d649df21df90a0371ffd1a0106fe17c6b1a56c1332b

**Review by miiu96 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**